lang/LangPrimSource/OSCData.cpp

```diff
@@ -925,6 +926,9 @@ void cleanup_OSC() {
     postfl("cleaning up OSC\n");
 
     stopAsioThread();
+    // destroy socket after ASIO thread has been joined!
```

Comment:
Just regarding your comment: does `stopAsioThread();` join the thread? I would have thought it just ends it.

Reply:
The existing function joins the thread

https://github.com/capital-G/supercollider/blob/c87ec0782b351a47b24f23ff3a8b454b7873715c/lang/LangPrimSource/SC_ComPort.cpp#L62-L65

I'll maybe try to refactor the asio thread into its own class so we don't mangle with global variables from two files which makes it really scattered and hard to follow, but this would be another PR.